src/Symfony/Component/DependencyInjection/EnvironmentBag/EnvironmentBag.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\EnvironmentBag;
+
+use Symfony\Component\DependencyInjection\Exception\EnvironmentVariableNotFoundException;
+
+/**
+ * Environment bag that uses $_SERVER to get environment variables.
+ *
+ * @author Magnus Nordlander <magnus@fervo.se>
+ */
+class EnvironmentBag implements EnvironmentBagInterface
+{
+    protected $imprintedVars;
+
+    /**
+     * @param array|null $allowedVars An array of allowed environment variables or null if all variables are allowed.
+     */
+    public function __construct(array $whitelist = null)
+    {
+        if ($whitelist) {
+            $this->imprintWhitelist($whitelist);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function imprintWhitelist(array $whitelist)
+    {
+        $this->imprintedVars = [];
+        foreach ($whitelist as $name) {
+            $value = $this->resolveEnvironmentVariable($name);
+            if ($value) {
+                $this->imprintedVars[$name] = $value;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isImprinted()
+    {
+        return $this->imprintedVars !== null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function has($name)
+    {
+        if ($this->isImprinted($name)) {
+            return isset($this->imprintedVars[$name]);
+        }
+
+        return $this->resolveEnvironmentVariable($name) !== null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($name)
+    {
+        if ($this->isImprinted($name) && isset($this->imprintedVars[$name])) {
+            return $this->imprintedVars[$name];
+        } elseif ($value = $this->resolveEnvironmentVariable($name) !== null) {
+            return $value;
+        }
+
+        throw new EnvironmentVariableNotFoundException($name);
+    }
+
+    protected function resolveEnvironmentVariable($name)
+    {
+        if (isset($_ENV[$name])) {
+            return $_ENV[$name];
+        }
+
+        if ($value = getenv($name)) {
+            return $value;
+        }
+
+        if (isset($_SERVER[$name])) {
```

Comment:
That whole section can go. If it's not returned by `getenv()`, it won't be in `$_SERVER` either. Furthermore, this code would allow retrieval of `HTTP_USER_AGENT` as an env var.